tests/integration/large_deployments/test_orchestration.py

```diff
@@ -0,0 +1,353 @@
+        ),
+        ops_test.model.deploy(
+            charm,
+            application_name=_2nd_DATA_APP,
+            num_units=APP_UNITS[_2nd_DATA_APP],
+            series=series,
+            config={"cluster_name": CLUSTER_NAME, "roles": "data", "init_hold": True}
+            | CONFIG_OPTS,
+            storage={"opensearch-data": "local,128G,1"},
+        ),
+    )
+    for app in APP_UNITS:
+        await ops_test.model.integrate(app, TLS_CERTIFICATES_APP_NAME)
+
+    for app in [FAILOVER_APP, DATA_APP, _2nd_DATA_APP]:
+        await ops_test.model.integrate(
+            f"{MAIN_APP}:peer-cluster-orchestrator", f"{app}:peer-cluster"
+        )
+
+    for app in [DATA_APP, _2nd_DATA_APP]:
+        await ops_test.model.integrate(
+            f"{FAILOVER_APP}:peer-cluster-orchestrator", f"{app}:peer-cluster"
+        )
+
+    await wait_until(
+        ops_test,
+        apps=[MAIN_APP, DATA_APP, FAILOVER_APP, _2nd_DATA_APP, TLS_CERTIFICATES_APP_NAME],
+        apps_statuses=["active"],
+        units_full_statuses={
+            MAIN_APP: {"units": {"active": []}},
+            DATA_APP: {"units": {"active": []}},
+            FAILOVER_APP: {"units": {"active": []}},
+            _2nd_DATA_APP: {"units": {"active": []}},
+            TLS_CERTIFICATES_APP_NAME: {"units": {"active": []}},
+        },
+        wait_for_exact_units=1,
+    )
+
+
+@pytest.mark.abort_on_fail
+async def test_check_orchestrators(ops_test: OpsTest) -> None:
+    """Test that the orchestrators are correctly set."""
+    assert ops_test.model is not None
+
+    data_app = ops_test.model.applications[DATA_APP]
+    assert data_app is not None
+    orchestrators = await get_application_relation_data(
+        ops_test,
+        unit_name=data_app.units[0].name,
+        relation_name=PeerRelationName,
+        key="orchestrators",
+    )
+    assert orchestrators, "No orchestrators found in relation data"
+    orchestrators = PeerClusterOrchestrators.from_dict(json.loads(orchestrators))
+    assert (
+        orchestrators.main_app and orchestrators.main_app.name == MAIN_APP
+    ), "Main orchestrator not set correctly"
+    assert (
+        orchestrators.failover_app and orchestrators.failover_app.name == FAILOVER_APP
+    ), "Failover orchestrator not set correctly"
+
+
+@pytest.mark.abort_on_fail
+async def test_demotion_through_relation_removal(ops_test: OpsTest) -> None:
+    assert ops_test.model is not None
+
+    main_app = ops_test.model.applications[MAIN_APP]
+    assert main_app is not None
+
+    for app in [FAILOVER_APP, DATA_APP, _2nd_DATA_APP]:
+        await main_app.remove_relation(
+            f"{MAIN_APP}:peer-cluster-orchestrator", f"{app}:peer-cluster"
+        )
+
+    await wait_until(
+        ops_test,
+        apps=[MAIN_APP, DATA_APP, FAILOVER_APP, _2nd_DATA_APP],
+        apps_full_statuses={
+            MAIN_APP: {"active": []},
+            FAILOVER_APP: {"blocked": [CMS_NOT_ENOUGH_STATUS]},
```

Comment:
Please remind me of this one, wasn't that taken out during the Profiles rework? meaning unless the profile was set to production this should not be set